### PR TITLE
Fixed converter

### PR DESCRIPTION
**Pull request by ybs007**

### CODE REVIEW

File: Blizzard.IO.RabbitMQ/RabbitPropertiesConverter.cs [hunk 1]
```diff
@@ -12,23 +10,84 @@ public class RabbitPropertiesConverter :
...
+            }
+
+            if (rabbitMessageProperties.Headers != null)
+            {
+                messageProperties.Headers = rabbitMessageProperties.Headers;
+            }
+            else
+            {
+                // By default the headersPresent is set to true
+                messageProperties.HeadersPresent = false;
```
Comment (alloncm) [inline Blizzard.IO.RabbitMQ/RabbitPropertiesConverter.cs]: Why do you need this `else`?

File: Blizzard.IO.RabbitMQ/RabbitPropertiesConverter.cs [hunk 2]
```diff
@@ -12,23 +10,84 @@ public class RabbitPropertiesConverter :
     {
         public MessageProperties Convert(RabbitMessageProperties rabbitMessageProperties)
```
Comment (alloncm) [inline Blizzard.IO.RabbitMQ/RabbitPropertiesConverter.cs:11]: Maybe we should just use the first method and assign to the present booleans true or false based on the null?

File: Blizzard.IO.RabbitMQ/RabbitPropertiesConverter.cs (hunk 1 above)
Comment (ybs007) [reply Blizzard.IO.RabbitMQ/RabbitPropertiesConverter.cs]: Fixed

File: Blizzard.IO.RabbitMQ/RabbitPropertiesConverter.cs (hunk 2 above)
Comment (ybs007) [reply Blizzard.IO.RabbitMQ/RabbitPropertiesConverter.cs:11]: Fixed

File: Blizzard.IO.RabbitMQ/Entities/RabbitMessageProperties.cs [hunk 3]
```diff
@@ -1,25 +1,23 @@
...
         public string MessageId { get; set; }
         public string CorrelationId { get; set; }
         public string ReplyTo { get; set; }
-        public long Timestamp { get; set; }
+        public long? Timestamp { get; set; }
         public string UserId { get; set; }
         public string AppId { get; set; }
         public string ClusterId { get; set; }
         public string Expiration { get; set; }
-        public bool Persistent { get; set; }
```
Comment (alloncm) [inline Blizzard.IO.RabbitMQ/Entities/RabbitMessageProperties.cs:22]: Why did you removed this?

File: Blizzard.IO.RabbitMQ/RabbitPropertiesConverter.cs [hunk 4]
```diff
@@ -12,22 +10,36 @@ public class RabbitPropertiesConverter :
...
+                Timestamp = rabbitMessageProperties.Timestamp ?? 0,
+                TimestampPresent = rabbitMessageProperties.Timestamp != null,
+                Priority = rabbitMessageProperties.Priority ?? 0,
+                PriorityPresent = rabbitMessageProperties.Priority != null,
                 UserId = rabbitMessageProperties.UserId,
-                AppId = rabbitMessageProperties.AppId,
-                ClusterId = rabbitMessageProperties.ClusterId,
+                UserIdPresent = rabbitMessageProperties.UserId != null,
                 Expiration = rabbitMessageProperties.Expiration,
-                Priority = rabbitMessageProperties.Priority
```
Comment (alloncm) [inline Blizzard.IO.RabbitMQ/RabbitPropertiesConverter.cs:30]: Also here

File: Blizzard.IO.RabbitMQ/Entities/RabbitMessageProperties.cs (hunk 3 above)
Comment (ybs007) [reply Blizzard.IO.RabbitMQ/Entities/RabbitMessageProperties.cs:22]: There is no such field in the message properties

File: Blizzard.IO.RabbitMQ/RabbitPropertiesConverter.cs (hunk 4 above)
Comment (ybs007) [reply Blizzard.IO.RabbitMQ/RabbitPropertiesConverter.cs:30]: Moved it up , it still exists

**Pull request merged**